src/pybind/mgr/snap_schedule/fs/schedule.py

```diff
@@ -397,14 +402,20 @@ def parse_schedule(cls, schedule: str) -> Tuple[int, str]:
     @property
     def repeat(self) -> int:
         period, mult = self.parse_schedule(self.schedule)
-        if mult == 'M':
+        if period <= 0:
+            raise ValueError("schedule period must be a non-zero positive value")
```

Comment:
While the exception string is correct, using something like `invalid schedule specified` is a bit more understandable to the user.

Reply:
My thoughts about `invalid schedule specified` are a bit contrary. With this, the user gets to know that things aren't right, but doesn't exactly get to know what is right.
One might argue that negative or zero period not being acceptable is pretty common sense and an `invalid period specified` should give enough hint. But I still prefer my version.

Do you still want me to make it `invalid schedule specified` ?

Reply:
Since this error throws up in the CLI, we'd need to make it non-cryptic and close to what the problem could be. Say the user passes `1mm`, the error message about non-zero value is a bit on the confusing side (but if `-1m` is passed, then its very much descriptive).

So, IMO, something that tells the user that the passed in schedule isn't correct would go a long way.

Reply:
How about:
```
invalid schedule specified - schedule period must be a non-zero positive value
```

Reply:
> How about:
> 
> ```
> invalid schedule specified - schedule period must be a non-zero positive value
> ```

Better, but IMO, we should hint the possible schedules accepted. Something like

```
invalid schedule specified  -- m,h,D,M,Y are valid schedules.
```

Reply:
ack; will update the exception messages in relevant places
found a couple of other places where the change from 'M' to 'm' was missing; will fix that
couldn't get the changes tested due to ongoing sepialab activity of OS upgrade

Reply:
> > How about:
> > ```
> > invalid schedule specified - schedule period must be a non-zero positive value
> > ```
> 
> Better, but IMO, we should hint the possible schedules accepted. Something like
> 
> ```
> invalid schedule specified  -- m,h,D,M,Y are valid schedules.
> ```

I think this change is for the co-efficient right? ` for `-1` in -1m`. I think milind's is much better,

`invalid schedule specified -  period must be a non-zero positive value` looks better

Reply:
> > > How about:
> > > ```
> > > invalid schedule specified - schedule period must be a non-zero positive value
> > > ```
> > 
> > 
> > Better, but IMO, we should hint the possible schedules accepted. Something like
> > ```
> > invalid schedule specified  -- m,h,D,M,Y are valid schedules.
> > ```
> 
> I think this change is for the co-efficient right? for `-1` in `-1m`. I think milind's is much better,
> 
> `invalid schedule specified - period must be a non-zero positive value` looks better

What about the _invalid_ 1MM? Is that negative or positive? :)

Reply:
> i think this is not for `MM`, that will be validated at `raise ValueError(f'schedule multiplier "{mult}" not recognized')`, this is for the prefix values i.e. 1 in case for 1MM

I have a couple of changes lined up and will push an update after testing.